File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int size = nums.length;

        int sum = 0;
        for(int i = 0 ; i < k ; i++)
            sum += nums[i];

        int maxSum = sum;

        for(int i = k ; i < size ; i++) {
            sum += nums[i];
            sum -= nums[i - k];
            maxSum = Math.max(sum, maxSum);
        }

        return (double) maxSum / k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        size, sum = len(nums), 0

        for i in range(k):
            sum += nums[i]

        maxSum = sum

        for i in range(k, size):
            sum += nums[i]
            sum -= nums[i - k]
            maxSum = max(maxSum, sum)

        return float(maxSum) / k